app/services/feeds/import.rb

```diff
@@ -52,9 +63,15 @@ def call
 
     private
 
-    attr_reader :earlier_than, :users_batch_size, :num_fetchers, :num_parsers
+    attr_reader :earlier_than, :users_batch_size, :num_fetchers, :num_parsers, :users
+
+    # TODO: jeremyf, the logic around which users is a bit convoluted; now we need to consider Article policies
+    #
+    # using nil here to avoid an unnecessary table count to check presence
+    def set_users_from(users_scope:, earlier_than:)
+      @users = users_scope.where(id: Users::Setting.with_feed.select(:user_id))
```

Comment:
I really like this move. The fact that ActiveRecord automatically writes nested SQL statements is so nice here.

```sql
SELECT "users".*
FROM "users"
WHERE "users"."id" IN (
  SELECT "users_settings"."id"
  FROM "users_settings"
  WHERE NOT (
    "users_settings"."feed_url" = ''
    OR "users_settings"."feed_url" IS NULL
  )
)
```